src/components/templates/FireBarrel/FireBarrel.tsx

```diff
@@ -90,8 +92,12 @@ const FireBarrel: React.FC = () => {
     [updateVideoState, profile]
   );
 
-  return useMemo(
-    () => (
+  return useMemo(() => {
+    if (!currentPartygoers) {
+      return <LoadingPage />;
+    }
```

Comment:
[nit] you can use single line guard statement

```suggestion
    if (!currentPartygoers) return <LoadingPage />;
```